src/ucp/wireup/wireup_cm.c

```diff
@@ -733,21 +737,28 @@ static void ucp_cm_server_connect_cb(uct_ep_h ep, void *arg,
     uct_worker_cb_id_t prog_id = UCS_CALLBACKQ_ID_NULL;
     ucp_lane_index_t cm_lane;
 
-    if (status != UCS_OK) {
-        /* if reject is arrived on server side, then UCT does something wrong */
-        ucs_assert(status != UCS_ERR_REJECTED);
+    if (status == UCS_OK) {
+        uct_worker_progress_register_safe(ucp_ep->worker->uct,
+                                          ucp_cm_server_connect_progress,
+                                          ucp_ep, UCS_CALLBACKQ_FLAG_ONESHOT,
+                                          &prog_id);
+        ucp_worker_signal_internal(ucp_ep->worker);
+    } else if (status == UCS_ERR_CONNECTION_RESET) {
+        /* remote side initiated disconnect before local side has completed
+         * connection establishment, so:
+         * 1) establish connection to complete local flush
```

Comment:
when was the local flush started?

Reply:
the EP can be closed (buggy case), ucp_ep_close_nb starts internal flash which will be completed by flushing transport lanes + by remote disconnect_cb.
thanks for pointing! will explain better in the comment.